packages/chain-adapters/src/cosmossdk/osmosis/OsmosisChainAdapter.ts

```diff
@@ -1,19 +1,32 @@
-import { ASSET_REFERENCE, AssetId, fromChainId, toAssetId } from '@shapeshiftoss/caip'
-import { bip32ToAddressNList, OsmosisSignTx, supportsOsmosis } from '@shapeshiftoss/hdwallet-core'
+import {
+  ASSET_REFERENCE,
+  AssetId,
+  CHAIN_REFERENCE,
+  fromChainId,
+  toAssetId
+} from '@shapeshiftoss/caip'
+import {
+  bip32ToAddressNList,
+  OsmosisSignTx,
+  OsmosisTx,
+  supportsOsmosis
+} from '@shapeshiftoss/hdwallet-core'
 import { BIP44Params, chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+import { bech32 } from 'bech32'
 
-import { ChainAdapter as IChainAdapter } from '../../api'
 import { ErrorHandler } from '../../error/ErrorHandler'
 import { toPath } from '../../utils'
+import { bnOrZero } from '../../utils/bignumber'
 import { ChainAdapterArgs, CosmosSdkBaseAdapter } from '../CosmosSdkBaseAdapter'
 
-export class ChainAdapter
-  extends CosmosSdkBaseAdapter<ChainTypes.Osmosis>
-  implements IChainAdapter<ChainTypes.Osmosis>
-{
+export class ChainAdapter extends CosmosSdkBaseAdapter<ChainTypes.Osmosis> {
   protected readonly supportedChainIds = ['cosmos:osmosis-1', 'cosmos:osmo-testnet-1']
```

Comment:
we shouldn't include supported chain ids that aren't in the `supportedChainIds`. We could also type this variable with `SupportedChainIds` to ensure we don't add chain ids that we don't support
```suggestion
  protected readonly supportedChainIds = ['cosmos:osmosis-1']
```

Reply:
Applicable to the other chain adapters as well